### PR TITLE
Add firmware upload  and trace enablement functionality to the HwModel trait

### PR DESCRIPTION
**Pull request by rusty1968**
The intent of this commit is to extend the functionality exposed by the HwModel trait.

1. Make the basic flow of uploading  Caliptra firmware into Mailbox SRAM buffer available to test suites in the caliptra_hw_model crate. 
2.  Expose the trace enablement interface.

### CODE REVIEW

File: hw-model/src/model_emulated.rs [hunk 1]
```diff
@@ -91,4 +91,8 @@ impl crate::HwModel for ModelEmulated {
 
         &mut self.output
     }
+
+    fn tracing_hint(&mut self, _enable: bool) {
```
Comment (korran) [inline hw-model/src/model_emulated.rs:95]: These tracing changes aren't mentioned in the PR description. Perhaps you intend this to be a separate PR?

File: hw-model/src/lib.rs [hunk 2]
```diff
@@ -216,4 +281,46 @@ mod tests {
             "expected output \"ha\", was \"hi\""
         );
     }
+
+    #[test]
+    pub fn test_upload_firmware() {
+        let firmware: Vec<u8> = [
```
Comment (korran) [inline hw-model/src/lib.rs:287]: Can you add a comment describing what this firmware does?

Comment (korran) [reply hw-model/src/lib.rs:287]: Oh wait, this doesn't appear to be used

Comment (rusty1968) [reply hw-model/src/lib.rs:287]: This is a test routine for the upload_firmware method. I can add the comment.

File: hw-model/src/model_emulated.rs (hunk 1 above)
Comment (rusty1968) [reply hw-model/src/model_emulated.rs:95]: Sure.

File: hw-model/src/lib.rs (hunk 2 above)
Comment (korran) [reply hw-model/src/lib.rs:287]: Oh, sorry, this is separate from the ROM used above, I'm crazy.

**Review by korran — APPROVED** [review]

**Review by korran — APPROVED** [review]
